interface/modules/custom_modules/oe-module-weno/src/Services/DownloadWenoPharmacies.php

```diff
@@ -12,11 +12,149 @@
 
 namespace OpenEMR\Modules\WenoModule\Services;
 
+if (!$GLOBALS ?? null) {
+    require_once dirname(__DIR__, 5) . "/globals.php";
+}
+
+use Exception;
+use League\Csv\Reader;
+use League\Csv\Statement;
+use mysqli;
 use OpenEMR\Common\Logging\EventAuditLogger;
 use ZipArchive;
 
 class DownloadWenoPharmacies
 {
+    public function __construct()
+    {
+    }
+
+    /**
+     * @param      $filePath
+     * @param bool $isInsertOnly
+     * @return false|int
+     */
+    public function processWenoPharmacyCsv($filePath, bool $isInsertOnly = true): false|int
+    {
+        $wenoLog = new WenoLogService();
+
+        if (date("l") == "Monday" && $isInsertOnly) {
+            $sql = "TRUNCATE TABLE weno_pharmacy";
+            sqlStatement($sql);
+        }
+
+        // Database configuration. Create connection.
+        $connect = new mysqli($GLOBALS["host"], $GLOBALS["login"], $GLOBALS["pass"], $GLOBALS["dbase"]);
```

Comment:
Just checking why you needed to do use direct mysqli calls.
Couple issues that will come up (can be dealt with with example in how done in setup.php script):
1. Setting of encoding
2. Breaking if mysql is set up to enforce encryption over wire

Reply:
Actually, code is in Installer.class.php and not setup.php

Reply:
yeah I simply forgot to go back to setting names. I was going to used same as we used in portal and got side tracked but you're correct this can cause issues that could be very hard to track down.

The reasons for direct management are of concern for me and something we need to address. 1. Our wrappers are slow. 2. If I was just importing from UI I wouldn't care too much however I've found out how easy running a database intensive script in background services can cause problem for front facing foreground runtime. It can hang foreground enough to be noticed by user where they can start clicking randomly while trying to figure out why app is hung.

The performance difference is quit noticeable by a factor of 50% which I benched marked.

I'm going to once again warn everyone that we need to take a close look at what we run in these background tasks.
My experience looking at a couple production sites that have several modules turn on plus our house keeping tasks like the UUID update, direct email polling, X12 severely affect openemr runtime from UI POV. 

What I've done here for this query intensive transaction needs to be an example of doing the same for some of our other queries that affect performance.

Having our current wrappers are/were a nice convenience for the type of workloads expected 10 years ago but as openemr grows with far greater data processing requirements and moving more to the enterprise, we have to/must put an effort into profiling and tuning openemr if we are going to compete for large enterprise requirements.
I haven't even brough into the affect API's will have as well.

I just don't want to be the only admin that's investigating this stuff or not worried about. We have to get used to having much larger data sets for testing with more realistic options turned on.

Anyway thanks for the catch Brady. I knew about but simply forgot!

Reply:
@bradymiller Can you help me out with this? Not sure best way to open connection and my brain is so tired...

Reply:
never mind @bradymiller I found a way that's an easy compromise. Going to use the existing connection from globals